.github/workflows/create-release.yaml

```diff
@@ -0,0 +1,69 @@
+name: "Create release"
+run-name: "Create release ${{ inputs.release_version }}"
+
+permissions:
+  contents: write
+
+on:
+  workflow_dispatch:
+    inputs:
+      release_version:
+        description: "Release version (i.e. x.y.z)"
+        default: ""
+        required: true
+      dry-run:
+        type: boolean
+        description: "Creates a dry-run branch and tag. This will not trigger the actual release process."
+        default: true
+
+jobs:
+  # check if tag is a valid semantic version
+  # check that no release already exists with that name
+  validate-release:
+    name: Validate release
+    runs-on: ubuntu-latest
+    steps:
+      # checkout code including all history
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: main
+
+      - name: Check if that release tag is correctly formatted
+        run: "./hack/scripts/check_release_tag.sh ${{ inputs.release_version }}"
+
+      - name: Check if release tag already exists
+        run: |
+          if [ $(git tag -l ${{ inputs.release_version }}) ]; then
+            echo "::error ::Tag ${{ inputs.release_version }} already exists"
+            exit 1
+          fi
+
+      - name: Check for existing artifacts from previous release runs
+        run: "./hack/scripts/check_artifacts_existence.sh ${{ inputs.release_version }}"
+
+      # create a branch for the release and push it to the repository
+      - name: Create release branch
+        if: ${{ inputs.dry_run == false }}
+        run: |
+          git checkout -b release-${{ inputs.release_version }}
+          git push origin release-${{ inputs.release_version }}
+
+      - name: Create release branch
```

Comment:
Maybe you can add a "(dry-run)" prefix to the action name to make it more transparent that the action is doing something different then usual